ipod/merge.py

```diff
@@ -0,0 +1,257 @@
+    reconsider_chi2: float = 8.0,
+    max_iter: int = 10,
+    min_mjd: Optional[float] = None,
+    max_mjd: Optional[float] = None,
+    database_directory: str = "",
+    datasets: Optional[set[str]] = None,
+    propagator: Type[Propagator] = PYOORB,
+    propagator_kwargs: dict = {},
+    chunk_size: int = 100,
+    max_processes: Optional[int] = 1,
+) -> Tuple[FittedOrbits, FittedOrbitMembers, PrecoveryCandidates, SearchSummary]:
+    time_start = time.perf_counter()
+    logger.info("Running merge and extend...")
+
+    orbits_out = FittedOrbits.empty()
+    orbit_members_out = FittedOrbitMembers.empty()
+    orbit_candidates_out = PrecoveryCandidates.empty()
+    search_summary_out = SearchSummary.empty()
+
+    if max_processes is None:
+        max_processes = mp.cpu_count()
+
+    # Initialize ray cluster or connect to an
+    # existing ray cluster
+    initialize_use_ray(num_cpus=max_processes)
+
+    orbits_iter = orbits
+    orbit_members_iter = orbit_members
+    observations_iter = observations
+
+    # Drop duplicate orbits (orbits with identical state vectors)
+    num_orbits = len(orbits_iter)
+    orbits_iter, orbit_members_iter = drop_duplicate_orbits(
+        orbits_iter, orbit_members_iter
+    )
+    logger.info(
+        f"Dropped {num_orbits - len(orbits_iter)} state vector duplicate orbits."
+    )
+    num_orbits = len(orbits_iter)
+
+    iterations = 0
+    while len(orbits_iter) > 0:
+        logger.info(f"Starting iteration {iterations + 1} of {max_iter}.")
+
+        chunk_size_iter = np.minimum(
+            np.ceil(len(orbits_iter) / max_processes).astype(int), chunk_size
+        )
+
+        # Run iterative precovery and differential correction
+        (
+            orbits_iter,
+            orbit_members_iter,
+            precovery_candidates_iter,
+            search_summary_iter,
+        ) = iterative_precovery_and_differential_correction(
+            orbits_iter,
+            orbit_members=orbit_members_iter,
+            observations=observations_iter,
+            max_tolerance=max_tolerance,
+            delta_time=delta_time,
+            rchi2_threshold=rchi2_threshold,
+            outlier_chi2=outlier_chi2,
+            reconsider_chi2=reconsider_chi2,
+            max_iter=10,
+            min_mjd=min_mjd,
+            max_mjd=max_mjd,
+            database_directory=database_directory,
+            datasets=datasets,
+            propagator=propagator,
+            propagator_kwargs=propagator_kwargs,
+            chunk_size=chunk_size_iter,
+            max_processes=max_processes,
+        )
+
+        orbits_iter, orbit_members_iter = assign_duplicate_observations(
+            orbits_iter, orbit_members_iter
+        )
+        if orbits_iter.fragmented():
+            orbits_iter = qv.defragment(orbits_iter)
+        if orbit_members_iter.fragmented():
+            orbit_members_iter = qv.defragment(orbit_members_iter)
+
+        precovery_candidates_unique = drop_duplicates(
```

Comment:
I'm assuming duplicate precovery candidates would have come from orbits that had observations dropped in `assign_duplicate_obsevations`. Since precovery candidates have residuals associated to them relative to an orbit, do we need to select the precovery candidates specific to the orbits that duplicate observations were assigned to?